crates/router/src/core/payments/transformers.rs

```diff
@@ -516,9 +516,10 @@ impl<F: Clone> TryFrom<PaymentAdditionalData<'_, F>> for types::PaymentsAuthoriz
         ));
 
         Ok(Self {
+            // payment_method_data is not required during recurring mandate payment, in such case keep default as PaymentMethodData as MandatePayment
             payment_method_data: payment_data
                 .payment_method_data
-                .get_required_value("payment_method_data")?,
+                .unwrap_or(api_models::payments::PaymentMethodData::MandatePayment),
```

Comment:
Hey, here you are defaulting, the payment_method_data to `MandatePayment` in case when this is not a recurring payment and the data is `None`, this transformation will go through and might cause inconsistent state in the steps that follow. 
Can we have a check of some sorts here, which would insure this behaviour is validated and error in thrown in the invalid cases?

Reply:
Addressed